source/features/render-conversation-template.tsx

```diff
@@ -0,0 +1,92 @@
+import * as textFieldEdit from 'text-field-edit';
+
+import features from '.';
+
+const placeholder = '%%%#%%%';
+
+/** Render using GitHub’s "Preview" feature */
+async function markdownToHTML(markdown: string): Promise<string> {
+	const token = select<HTMLInputElement>('.js-data-preview-url-csrf')!.value;
+
+	const body = new FormData();
+	body.set('text', markdown);
+	body.set('authenticity_token', token);
+
+	const url = select('[data-preview-url]')!.dataset.previewUrl!;
+	const response = await fetch(url, {body, method: 'post'});
+	return response.text();
+}
+
+function generateForm(html: string): Element {
+	// The classes try to match the spacing from `originalField`
+	const renderedTemplate = <div className="rgh-render-conversation-template markdown-body mx-0 pt-2 mb-2 mx-md-2 border-top"/>;
+	renderedTemplate.innerHTML = html.replaceAll(placeholder, '<textarea class="form-control input-contrast"></textarea>');
+
+	// Move the text immediately following the field to the field’s placeholder
+	for (const field of select.all('textarea', renderedTemplate)) {
+		field.placeholder = field.parentElement!.textContent!.trim();
+		while (field.nextSibling) {
+			field.nextSibling.remove();
+		}
+	}
+
+	return renderedTemplate;
+}
+
+async function init(): Promise<void | false> {
+	const originalField = select<HTMLTextAreaElement>('.js-comment-field')!;
+
+	const [preservePRText, template] = originalField.value.split('<!-- Please follow the template -->');
+	if (!template) {
+		return false;
+	}
+
+	// This will let us place the textarea’s values back into this string before submission
+	const templateWithPlaceholders = template
+		.replaceAll(/:[*_\s]+\n/g, `$&\n${placeholder}`);
+
+	// Makes comments visible, but only in the rendered template
+	const renderableTemplate = templateWithPlaceholders
+		.replaceAll('<!--', '')
+		.replaceAll('-->', '');
+
+	const renderedTemplate = generateForm(await markdownToHTML(renderableTemplate));
+	select('markdown-toolbar')!.before(renderedTemplate);
+
+	// Set original, template-less text back into the field
+	textFieldEdit.set(originalField, preservePRText);
+
+	// Focus first textarea
+	select('textarea', renderedTemplate)!.focus();
+
+	// Merge textareas
+	// TODO: Do this every time the preview is generated (before it's sent to GitHub)
+	originalField.form!.addEventListener('submit', () => {
+		const fields = select.all('textarea', renderedTemplate);
+		let counter = 0;
+		const filledTemplate = templateWithPlaceholders
+			.replaceAll(placeholder, () => fields[counter++].value);
+
+		originalField.value = filledTemplate + '\n\n' + originalField.value;
+
+		// Once the text is visibly merged for submission, the rendered template is no longer necessary
+		renderedTemplate.remove();
+	});
+}
+
+void features.add({
+	id: __filebasename
+}, {
+	include: [
+		pageDetect.isNewIssue,
+		pageDetect.isQuickPR,
+		// TODO: Merge with above
+		() => pageDetect.isCompare() && location.search.includes('expand=1')
```

Comment:
cc @yakov116 I think we need a unified `isNewPR`

Reply:
:+1: I can send a pr tomorrow. 

Does expand always happen on new pr's? I always used a dom based detection.

Reply:
Not sure. Here's an example URL:

```
https://github.com/sindresorhus/refined-github/compare/incremental-tag-changelog-link?expand=1
```

Reply:
https://github.com/sindresorhus/refined-github/compare/master...master?expand=1

This does not mean that I can create a PR. I think this is why we called it isCompare since we cant tell if its a new PR.

Reply:
> This does not mean that I can create a PR

True, but that's also true for `isQuickPR` if a PR already exists.

Should I just use `isCompare`?